libmachine/provision/debian.go

```diff
@@ -110,9 +111,13 @@ func (provisioner *DebianProvisioner) Provision(swarmOptions swarm.Options, auth
 		}
 	}
 
-	log.Debug("installing docker")
-	if err := installDockerGeneric(provisioner, engineOptions.InstallURL); err != nil {
-		return err
+	if strings.EqualFold(engineOptions.InstallURL, "none") {
```

Comment:
removing the yum update you can put this in installDockerGeneric and have one check